src/trade/selectors.js

```diff
@@ -0,0 +1,181 @@
+import { createSelector } from 'reselect';
```

Comment:
I actually prefer to group things by feature over layer, so that when we want to change something related to a feature, it's easier to browse in the same directory ~